example/chat/chat_test.go

```diff
@@ -13,40 +12,70 @@ import (
 func TestChat(t *testing.T) {
 	srv := httptest.NewServer(handler.GraphQL(MakeExecutableSchema(New())))
 	c := client.New(srv.URL)
-	var wg sync.WaitGroup
-	wg.Add(1)
 
 	t.Run("subscribe to chat events", func(t *testing.T) {
 		t.Parallel()
 
 		sub := c.Websocket(`subscription { messageAdded(roomName:"#gophers") { text createdBy } }`)
 		defer sub.Close()
 
-		wg.Done()
-		var resp struct {
+		postErrCh := make(chan error)
+		go func() {
+			var resp interface{}
+			// can't call t.Fatal from separate goroutine, so we return to the err chan for later
```

Comment:
You can't call fatal (because it relies on panic handlers...) but you can call t.Error and friends (assert.Equal, instead of require.Equal).

I think this can go back to the way it was, except using `go func`, and `assert` instead of `t.Run` and `require.`

Reply:
Ah, I think that comment might be a little misleading (see: oops forgot to update).

The issue we saw locally (after moving out of `t.Run`) was that `sub.Next` would hang indefinitely (or until timeout) if the mutation call failed (we did a negative test by making the mutation invalid to see) because it never gets any message.

The real reason it's there (and what that comment probably *should* say) is so we can use `select` to wait for either `sub.Next` *or* the `Post` call to fail and then exit to avoid the deadlock.